### ISSUE TITLE
Bug: sc-im doesn't respect terminal background and is blinking

### ISSUE THREAD

**Issue by 1dancook** [andmarti1424/sc-im#627]
### A little explanation

I'm just trying to get set up so I can switch between 0.8.2 and the 0.8.3 dev versions easily in order to test bugs and other things. 

Without modifying anything from the repo clone, doing the make steps, sc-im has some very strange behaviour for color and blinking. You can see in the attached .gif below. This is loaded without my scimrc and in macOS Terminal (didn't catch the blinking until I loaded it there!)

![screencap_sc-im-blinking](https://user-images.githubusercontent.com/12954427/142138645-b0b00046-e9d8-4d0b-82a7-2592e31a26cd.gif)

### More info
- this is not a problem with the home-brew install on macOS (no blinking either)
- I tried various versions from the git clone: 0.8.2, 0.8.3 main, 0.8.3 dev and this problem persists in all of them
- I also tried using the 0.8.2 zip from GitHub release but has the same issue

Some resolution would be appreciated so I can start testing out other things - this is making it a little difficult!

**Comment by andmarti1424** [comment]
Does this happen to you in other terminal?
Seems a terminal issue with ncurses. I believe you could redefine colors to avoid this problem.

**Comment by 1dancook** [comment]
I tried this in 3 different terminals. In macOS Terminal there is blinking. In iTerm2 there is no blinking but the white background is there for table cells and etc. In Kitty terminal (which is what I usually use) it's the same as iTerm2. I have all the colors defined in Kitty so I'm not sure really why this is happening.

> I believe you could redefine colors to avoid this problem.

Do you mean in scimrc?

**Comment by andmarti1424** [comment]
Yes indeed. Try the dracula theme. Its made by @reportaman who might use the same terminal as you do.

**Comment by 1dancook** [comment]
@andmarti1424 yes that worked! 🙂👍

With 0.8.3 dev branch the whole of sc-im is still blinking in Terminal app. Other terminal is fine. This could just be something on my end however, but I'm drawing blanks at the moment.  

---

Regarding colors: using a theme that redefines all colors works, however, the default settings of sc-im should not have this kind of problem. Will leave this open for now since this is a problem in 0.8.3 dev.

**Comment by 1dancook** [comment]
Using the dracula theme solved my color problem - but after trying to implement the colors that I had before, I may have gotten closer to the reason for the white background issues in the first place. 

Having `color "type=NORMAL fg=WHITE bg=BLACK" in my config will make this problem appear.

**Comment by 1dancook** [comment]
Last night I continued to try and reproduce my theme that I had set up from before. Having a color defined that should be gray would make things pink. White would make things yellow. All kinds of strangeness.

Dracula theme worked. But if I put a define_color statement before the redefine_color statements in that theme I would get a very different result. 

So that has led me to think that maybe something in the build process is wrong. Somehow 0.8.2 from home brew works fine. Building manually doesn't. 

I will try and investigate more and sorry for posting one comment above rather hastily- I thought I had found the culprit!

**Comment by 1dancook** [comment]
I'm going to close because I think this is not an sc-im issue. I don't know what the issue really is, though. I have compared the output of `brew -v install sc-im` with sc-im's `make` and the only difference I saw was that brew was using clang. I changed CC=clang but still the issue persists. 
I don't know if any other macOS users have encountered this problem but I'm willing to bet they probably haven't. It's likely some issue I have locally but hard to pin down.
